packages/typescript/src/index.js

```diff
@@ -58,6 +65,58 @@ module.exports = {
       },
     ],
 
+    // Recommended rules that require type information
+    '@typescript-eslint/no-unsafe-argument': 'off',
+    '@typescript-eslint/no-unsafe-assignment': 'off',
+    '@typescript-eslint/no-unsafe-call': 'off',
+    '@typescript-eslint/no-unsafe-member-access': 'off',
+    '@typescript-eslint/no-unsafe-return': 'off',
+    '@typescript-eslint/restrict-template-expressions': 'off',
```

Comment:
This rule prevents the use of non-`string` types in template expressions. We make use of this in certain places, like `key-tree` using `bip32:${number}` for derivation paths.

Reply:
Hmm, interesting. I like the idea of requiring explicit type casting. It's not too inconvenient, and can catch cases where you don't realize it's not a string.

Reply:
The thing is that we only want to accept numbers in those cases.

```ts
type PathSegment = `bip32:${number}`;

const foo: PathSegment = 'bip32:foo'; // Invalid
const bar: PathSegment = 'bip32:123'; // Valid
```

Reply:
Ohhh I misunderstood, that was a string template type not an actual string template with a variable called `number`.

Yeah that's a great point. Agreed with turning this off.

Reply:
Is it actually used for complex types, or only primitives? If the latter, maybe consider `allow-number` etc so we can eat the cake and have it without `[object Object]`s?

Reply:
Primitives and other types like this:

```ts
type Hex = `0x${string}`;
type Foo = `foo-${Hex}`;
```

I'm not sure how this rule handles the second case, I'll give it a try.

Reply:
Ah, I misunderstood the rule slightly. I think it's fine if we use `allowNumber`, `allowBoolean`, maybe `allowAny`?

Reply:
Hmm. Would we benefit from `allowAny`? The other two make sense.

Either way, having this rule enabled with `allowAny` would still be an improvement compared to today.

Reply:
Removed `allowAny`.